packages/block-library/src/search/index.php

```diff
@@ -132,8 +132,48 @@ function render_block_core_search( $attributes ) {
 			}
 		} else {
 			$button_classes[]       = 'has-icon';
+			$icon_dimensions        = '24';
+			$search_block_font_size = wp_get_global_styles(
+				array( 'typography', 'fontSize' ),
+				array(
+					'block_name' => 'core/search',
+					'transforms' => array( 'resolve-variables' ),
+				)
+			);
+
+			if ( ! empty( $attributes['style']['typography']['fontSize'] ) ) {
+				$icon_dimensions = esc_attr( $attributes['style']['typography']['fontSize'] );
+			} elseif ( ! empty( $attributes['fontSize'] ) ) {
+				$font_size_presets = wp_get_global_settings( array( 'typography', 'fontSizes' ) );
+				/*
+				 * The prioritization of origins matches the order in which the block editor fetches settings.
+				 * Font sizes are sourced from either 'typography.fontSizes.custom' || 'typography.fontSizes.theme' || 'typography.fontSizes.default'.
+				 * See: `getBlockSettings()` packages/block-editor/src/store/get-block-settings.js
+				 * @TODO Abstract the logic to get presets from origin into a block-supports/typography.php helper function.
+				 * @TODO Maybe update to use null coalescing operator when PHP 7.0 is the minimum version.
+				 */
+				$font_size_presets_from_origin = array();
+
+				if ( ! empty( $font_size_presets['custom'] ) ) {
+					$font_size_presets_from_origin = $font_size_presets['custom'];
+				} elseif ( ! empty( $font_size_presets['theme'] ) ) {
+					$font_size_presets_from_origin = $font_size_presets['theme'];
+				} elseif ( ! empty( $font_size_presets['default'] ) ) {
+					$font_size_presets_from_origin = $font_size_presets['default'];
+				}
+
+				if ( ! empty( $font_size_presets_from_origin ) ) {
+					$font_size_preset_index = array_search( $attributes['fontSize'], array_column( $font_size_presets_from_origin, 'slug' ) );
+					if ( isset( $font_size_presets_from_origin[ $font_size_preset_index ]['size'] ) ) {
+						$icon_dimensions = $font_size_presets_from_origin[ $font_size_preset_index ]['size'];
```

Comment:
Apologies if I'm overthinking it, but since the `custom` origin comes from the database rather than theme or core files, would it be worth adding an `esc_attr()` call to this line too?

Reply:
Not at all. It's best to sanitize late - thanks for pointing it out. 

To be honest, I'm in two minds about all this. It's an awful lot of code but I'm not sure there's another way, given that it should work for global styles as well 🤔

Reply:
> To be honest, I'm in two minds about all this. It's an awful lot of code but I'm not sure there's another way, given that it should work for global styles as well 🤔

That's a good point — what I like about this PR is that it's thorough about all the different possible cases for the size of the icon. Also, while it is a fair bit more code, it's guarded behind the `! $use_icon_button` check, so the code will only be executed when needed, which is good 👍

I can't think of a better or more streamlined way to do it on the PHP side. From my perspective, this PR is probably in good shape for gathering some wider feedback from folks to see whether the issue is worth the extra code?

Reply:
> I can't think of a better or more streamlined way to do it on the PHP side. From my perspective, this PR is probably in good shape for gathering some wider feedback from folks to see whether the issue is worth the extra code?

Thanks for the feedback 🙇🏻